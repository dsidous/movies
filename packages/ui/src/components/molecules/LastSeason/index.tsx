import React from 'react';
import { Link } from 'react-router-dom';

import { Season as ISeason } from '@typesRoots/tv';

import Season from '../Season';
import useStyles from './styles';

interface Props {
  tvId: number;
  season: ISeason;
}

const LastSeason: React.FC<Props> = ({ tvId, season }) => {
  const classes = useStyles();
  return (
    <div>
      <Season season={season} tvId={tvId} />
      <Link to={`/tv/${tvId}/seasons`} className={classes.allBtn}>
        View All Seasons
      </Link>
    </div>
  );
};

export default LastSeason;
