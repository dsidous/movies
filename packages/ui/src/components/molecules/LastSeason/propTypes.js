import { number, object } from 'prop-types';

export const propTypes = {
  tvId: number.isRequired,
  season: object.isRequired,
};

export const defaultProps = {};
