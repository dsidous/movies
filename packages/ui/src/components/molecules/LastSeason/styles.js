import { makeStyles } from '@material-ui/core/styles';

export default makeStyles(
  theme => ({
    root: {
      borderTop: theme.separator,
      margin: '40px auto 0',
      [theme.breakpoints.up('sm')]: {
        maxWidth: 960,
      },

      '& h4': {
        margin: '20px 10px',
      },
    },

    allBtn: {
      background: 'none',
      border: 'none',
      color: '#666',
      display: 'inline-block',
      fontSize: '1.2em',
      fontWeight: 'bold',
      margin: '20px 10px',
    },
  }),
  { name: 'lastSeason' },
);
