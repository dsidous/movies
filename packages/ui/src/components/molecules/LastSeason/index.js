import React from 'react';
import { Link } from 'react-router-dom';

import { propTypes } from './propTypes';
import Season from '../Season';

const LastSeason = ({ tvId, season }) => (
  <div className="last-season">
    <h4>Last season</h4>
    <Season season={season} tvId={tvId} />
    <Link to={`/tv/${tvId}/seasons`} className="all-seasons-btn">
      View All Seasons
    </Link>
  </div>
);

LastSeason.propTypes = propTypes;

export default LastSeason;
