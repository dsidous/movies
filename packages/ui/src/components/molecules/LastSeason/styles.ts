import { makeStyles } from '@material-ui/core/styles';

export default makeStyles(
  theme => ({
    allBtn: {
      display: 'inline-block',
      fontSize: '1.2em',
      fontWeight: 'bold',
      margin: theme.spacing(3, 0),
    },
  }),
  { name: 'lastSeason' },
);
