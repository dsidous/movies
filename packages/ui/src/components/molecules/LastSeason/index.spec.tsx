import React from 'react';
import { shallow } from 'enzyme';
import LastSeason from '.';

const mockProps = {
  tvId: 1,
  season: {
    season_id: 1,
  },
};

describe('Molecules/LastSeason', () => {
  it('should render as expected', () => {
    const wrapper = shallow(<LastSeason {...mockProps} />);

    expect(wrapper).toMatchSnapshot();
  });
});
