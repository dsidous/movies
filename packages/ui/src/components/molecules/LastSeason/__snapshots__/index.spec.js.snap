// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Molecules/LastSeason should render as expected 1`] = `ShallowWrapper {}`;
